sql/ttl/ttljob: TestRowLevelTTLJobRandomEntries failed

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6023773?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6023773?buildTab=artifacts#/) on master @ [24906bd59e975f607e9ea88cbd4ae5ff0432df45](https://github.com/cockroachdb/cockroach/commits/24906bd59e975f607e9ea88cbd4ae5ff0432df45):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries3126926523
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (52.48s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1071
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (10.72s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18438

Epic CRDB-14592

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6039299?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6039299?buildTab=artifacts#/) on master @ [6c24f3564e213371bbe800e9646386cdefeb1546](https://github.com/cockroachdb/cockroach/commits/6c24f3564e213371bbe800e9646386cdefeb1546):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries2755255190
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (50.86s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1038
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (12.17s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6039301?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6039301?buildTab=artifacts#/) on master @ [6c24f3564e213371bbe800e9646386cdefeb1546](https://github.com/cockroachdb/cockroach/commits/6c24f3564e213371bbe800e9646386cdefeb1546):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries3742589441
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (210.47s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1025
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (30.20s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6052227?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6052227?buildTab=artifacts#/) on master @ [fb8b849ba301a7b960a5f7d3b7a1bafffe51d106](https://github.com/cockroachdb/cockroach/commits/fb8b849ba301a7b960a5f7d3b7a1bafffe51d106):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries2401428305
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (55.73s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1056
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (10.45s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6068802?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6068802?buildTab=artifacts#/) on master @ [1ac67a9d1045f4111880156ca95dd4e5925ceece](https://github.com/cockroachdb/cockroach/commits/1ac67a9d1045f4111880156ca95dd4e5925ceece):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries342057656
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (217.30s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1059
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (31.84s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6068800?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6068800?buildTab=artifacts#/) on master @ [1ac67a9d1045f4111880156ca95dd4e5925ceece](https://github.com/cockroachdb/cockroach/commits/1ac67a9d1045f4111880156ca95dd4e5925ceece):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries3492974684
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (54.47s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1043
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (11.69s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6068805?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6068805?buildTab=artifacts#/) on master @ [1ac67a9d1045f4111880156ca95dd4e5925ceece](https://github.com/cockroachdb/cockroach/commits/1ac67a9d1045f4111880156ca95dd4e5925ceece):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries601951478
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (653.02s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1038
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (145.62s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6083429?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6083429?buildTab=artifacts#/) on master @ [dd59982d8f18d5aaba7643b5a2c3da782d0bcd2b](https://github.com/cockroachdb/cockroach/commits/dd59982d8f18d5aaba7643b5a2c3da782d0bcd2b):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries420987437
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (57.26s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1075
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (11.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6083431?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6083431?buildTab=artifacts#/) on master @ [dd59982d8f18d5aaba7643b5a2c3da782d0bcd2b](https://github.com/cockroachdb/cockroach/commits/dd59982d8f18d5aaba7643b5a2c3da782d0bcd2b):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries410201081
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (203.52s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1149
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (34.59s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6091286?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6091286?buildTab=artifacts#/) on master @ [d25cb57ccd9bc643ce9058ebd2057cab36b69ad5](https://github.com/cockroachdb/cockroach/commits/d25cb57ccd9bc643ce9058ebd2057cab36b69ad5):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries3029703061
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (50.86s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1015
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (10.18s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6091288?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6091288?buildTab=artifacts#/) on master @ [d25cb57ccd9bc643ce9058ebd2057cab36b69ad5](https://github.com/cockroachdb/cockroach/commits/d25cb57ccd9bc643ce9058ebd2057cab36b69ad5):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries966437385
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (167.20s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1012
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (29.32s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6098440?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6098440?buildTab=artifacts#/) on master @ [6b7f0058325de8b8b321a08137d462c418c3047f](https://github.com/cockroachdb/cockroach/commits/6b7f0058325de8b8b321a08137d462c418c3047f):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries833028037
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (187.35s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1057
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (35.64s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6112061?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6112061?buildTab=artifacts#/) on master @ [32f7d263043cfb47f3ac409a8aaa7346e56b973b](https://github.com/cockroachdb/cockroach/commits/32f7d263043cfb47f3ac409a8aaa7346e56b973b):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries2937061622
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (55.30s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1017
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (11.04s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6128033?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6128033?buildTab=artifacts#/) on master @ [f292c7a6f136af0b4f1cb9ee8de7af6c1107ac08](https://github.com/cockroachdb/cockroach/commits/f292c7a6f136af0b4f1cb9ee8de7af6c1107ac08):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries1020222568
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (54.37s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1092
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (12.66s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/ttl/ttljob.TestRowLevelTTLJobRandomEntries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6128038?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6128038?buildTab=artifacts#/) on master @ [f292c7a6f136af0b4f1cb9ee8de7af6c1107ac08](https://github.com/cockroachdb/cockroach/commits/f292c7a6f136af0b4f1cb9ee8de7af6c1107ac08):


```
=== RUN   TestRowLevelTTLJobRandomEntries
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/74bc9dd224da62240eee8570f9dfa5ee/logTestRowLevelTTLJobRandomEntries2627040363
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRowLevelTTLJobRandomEntries
    ttljob_test.go:685: -- test log scope end --
--- FAIL: TestRowLevelTTLJobRandomEntries (548.46s)
=== RUN   TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    ttljob_test.go:546: test case: ttljob_test.testCase{desc:"one column pk multiple nodes", createTable:"CREATE TABLE tbl (\n\tid UUID PRIMARY KEY DEFAULT gen_random_uuid(),\n\ttext TEXT\n) WITH (ttl_expire_after = '30 days')", preSetup:[]string(nil), postSetup:[]string(nil), numExpiredRows:1001, numNonExpiredRows:5, numSplits:10, forceNonMultiTenant:false, numNodes:5, expirationExpression:"", addRow:(func(*ttljob_test.rowLevelTTLTestJobTestHelper, *tree.CreateTable, time.Time))(nil)}
    ttljob_test.go:679: 
        	Error Trace:	ttljob_test.go:679
        	Error:      	Not equal: 
        	            	expected: 1001
        	            	actual  : 1063
        	Test:       	TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes
    --- FAIL: TestRowLevelTTLJobRandomEntries/one_column_pk_multiple_nodes (115.11s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRowLevelTTLJobRandomEntries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/ttl/ttljob/ttljob_processor.go
+++ b/pkg/sql/ttl/ttljob/ttljob_processor.go
@@ -36,6 +36,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/quotapool"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
@@ -73,7 +74,7 @@ func (ttl *ttlProcessor) work(ctx context.Context) error {
 		deleteRateLimit,
 	)
 
-	rowCount := int64(0)
+	processorRowCount := int64(0)
 
 	var relationName string
 	var pkColumns []string
@@ -151,7 +152,7 @@ func (ttl *ttlProcessor) work(ctx context.Context) error {
 						ttlSpec.PreSelectStatement,
 					)
 					// add before returning err in case of partial success
-					atomic.AddInt64(&rowCount, rangeRowCount)
+					atomic.AddInt64(&processorRowCount, rangeRowCount)
 					metrics.RangeTotalDuration.RecordValue(int64(timeutil.Since(start)))
 					if err != nil {
 						// Continue until channel is fully read.
@@ -224,18 +225,26 @@ func (ttl *ttlProcessor) work(ctx context.Context) error {
 		return err
 	}
 
-	job, err := serverCfg.JobRegistry.LoadJob(ctx, ttlSpec.JobID)
-	if err != nil {
-		return err
-	}
-	return db.Txn(ctx, func(_ context.Context, txn *kv.Txn) error {
-		return job.Update(ctx, txn, func(_ *kv.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
+	jobID := ttlSpec.JobID
+	return serverCfg.JobRegistry.UpdateJobWithTxn(
+		ctx,
+		jobID,
+		nil,  /* txn */
+		true, /* useReadLock */
+		func(_ *kv.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
 			progress := md.Progress
-			progress.Details.(*jobspb.Progress_RowLevelTTL).RowLevelTTL.RowCount += rowCount
+			existingRowCount := progress.Details.(*jobspb.Progress_RowLevelTTL).RowLevelTTL.RowCount
+			progress.Details.(*jobspb.Progress_RowLevelTTL).RowLevelTTL.RowCount += processorRowCount
 			ju.UpdateProgress(progress)
+			log.VInfof(
+				ctx,
+				2, /* level */
+				"TTL processorRowCount updated jobID=%d processorID=%d tableID=%d existingRowCount=%d processorRowCount=%d progress=%s",
+				jobID, ttl.ProcessorID, details.TableID, existingRowCount, processorRowCount, progress,
+			)
 			return nil
-		})
-	})
+		},
+	)
 }
 
 // rangeRowCount should be checked even if the function returns an error because it may have partially succeeded
@@ -348,13 +357,13 @@ func runTTLOnRange(
 				defer tokens.Consume()
 
 				start := timeutil.Now()
-				rowCount, err := deleteBuilder.run(ctx, ie, txn, deleteBatch)
+				batchRowCount, err := deleteBuilder.run(ctx, ie, txn, deleteBatch)
 				if err != nil {
 					return err
 				}
 
 				metrics.DeleteDuration.RecordValue(int64(timeutil.Since(start)))
-				rangeRowCount += int64(rowCount)
+				rangeRowCount += int64(batchRowCount)
 				return nil
 			}); err != nil {
 				return rangeRowCount, errors.Wrapf(err, "error during row deletion")

